### PR TITLE
Bug 1340544 - Automating the build process and using manual app groups/bundle IDs

### PR DESCRIPTION
**Pull request by sleroux**
Resolving the app group issue required me to make the profile management be manual and since I was messing around with that I figured I would take the opportunity to automate our Nightly/Beta releases better for the upcoming beta cycle.

1. I've moved in the scripts that were in firefox-ios-build-tools into a scripts folder
2. Added buddybuild_postclone.sh/buddybuild_prebuild.sh scripts which replicate what we do in fastlane - but on Buddybuild!
3. Removed the old Fennec scheme and renamed FennecEnterprise to Fennec.
4. Changed all of the bundle identifiers to use enterprise versions of our bundle identifiers. This is key in automating our beta/release builds on BB without handing over our distribution key for signing.
5. Turned on manual profile management with profiles created for Fennec/Nightly/Beta. 
6. Removed .xcconfig files since we don't really need them for much anymore cleaning up the build settings.

You can see a successful beta build (with localizations/badging and all) here:

https://dashboard.buddybuild.com/apps/57bf25c0f096bc01001e21e0/build/58adebdb4b5a9801009224e7

I'm working on a script that will allow us to resign the .IPAs that BB spits out with our distribution key so we can close the loop on automating our TestFlight submissions. PR to follow along with postbuild script to upload to our signing server.

### CODE REVIEW

File: Client/Info.plist [hunk 1]
```diff
@@ -3,15 +3,15 @@
 <plist version="1.0">
 <dict>
 	<key>AdjustAppToken</key>
-	<string>$(ADJUST_APP_TOKEN)</string>
+	<string></string>
 	<key>AdjustEnvironment</key>
-	<string>$(ADJUST_ENVIRONMENT)</string>
```
Comment (sleroux) [inline Client/Info.plist]: These are left empty and will be filled in during Buddybuilding

File: Client/Info.plist [hunk 2]
```diff
@@ -3,15 +3,15 @@
...
+	<string></string>
 	<key>AppIdentifierPrefix</key>
 	<string>$(APP_IDENTIFIER_PREFIX)</string>
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en</string>
 	<key>CFBundleDisplayName</key>
-	<string>$(MOZ_BUNDLE_DISPLAY_NAME)</string>
+	<string>$(DISPLAY_NAME)</string>
```
Comment (sleroux) [inline Client/Info.plist]: Moved the old display name property from the now-removed .xcconfig file to a user defined variable for this target.

File: Client/Info.plist (hunk 1 above)
Comment (farhanpatel) [inline Client/Info.plist]: "buddybuilding" I like that 👍  "How do you guys do your Ci?" "I've got a buddy.. he builds it."

**Comment by sleroux** [comment]
Closing this PR because BB freaked when I changed the base from master to v7.x.

**Pull request closed without merging**